### DISCUSSION TITLE
Is it possible to use mini.move with mini.pick?

### DISCUSSION THREAD

**Discussion by fnandogp** [nvim-mini/mini.nvim#1271] [Q&A] [answered]
Yesterday I set up `mini.pick` for the first time and love it.

So far, there is only one thing missing for me. Coming from Telescope (which I also love), I could use `<M-h>` and `<M-l>` from `mini.move` to move my cursor.

I don't understand much about the internals of how it works so it's hard to debug that a bit further. Overring keymaps was something I thought of but apparently, there are none.

Any ideas?

**Comment by echasnovski** [comment, marked as answer]
Welcome to the world of 'mini.pick' :)

> So far, there is only one thing missing for me. Coming from Telescope (which I also love), I could use `<M-h>` and `<M-l>` from `mini.move` to move my cursor.

Judging by your dotfiles, I would take a guess that you indeed mean `<M-h>` to `<M-l>` to move cursor left and right. This comes from ['mini.basics' `config.mappings.move_with_alt`](https://github.com/echasnovski/mini.nvim/blob/a535342b6105a684f16114f598d44318e6a1d20f/doc/mini-basics.txt#L313-L317) which [you indeed enable](https://github.com/fnandogp/dotfiles-macos/blob/f563bd33d68b77b95f38f24ec17ef3d2ef8adc99/config/nvim/lua/plugins/init.lua#L17). This is not the same as 'mini.move' which uses those keys by default in Normal and Visual modes (as opposed to just moving cursor in Insert, Command, and Terminal mode.

So with this in mind the answer: yes, you can customize which keys move cursor left and right. As 'mini.pick' implements its own [picker life cycle with its special keys](https://github.com/echasnovski/mini.nvim/blob/a535342b6105a684f16114f598d44318e6a1d20f/doc/mini-pick.txt#L195-L203) (and doesn't really have any real cursor active), you can override [`mappings.caret_left` and `mappings.caret_right`](https://github.com/echasnovski/mini.nvim/blob/a535342b6105a684f16114f598d44318e6a1d20f/doc/mini-pick.txt#L468-L469).

In practice this will look something like this:

```lua
require('mini.pick').setup({
  mappings = {
    caret_left = '<M-h>',
    caret_right = '<M-l>',
  }
})
```

Note that this means that moving left and right with `<Left>` and `<Right>` keys will be overridden / disabled.

**Reply by fnandogp** [reply]
That was it!
Thank you! And thank you again for `mini.nvim`.
